js/rebase-client/src/lib.rs

```diff
@@ -33,40 +34,56 @@ pub struct Client {
     client: Arc<RebaseClient>,
 }
 
+#[wasm_bindgen]
+#[derive(Deserialize, Serialize)]
+pub struct Config {
+    instructions: String,
+    statement: String,
+    jwt: Option<String>,
+    ld: Option<String>,
+    verify_jwt: Option<String>,
+    verify_ld: Option<String>,
+}
+
+impl Config {
+    fn is_valid(&self) -> Result<(), String> {
+        if self.jwt.is_none() && self.ld.is_none() {
+            Err("At least one of JWT or LD urls must be set".to_string())
+        } else {
+            Ok(())
+        }
+    }
+}
+
 #[wasm_bindgen]
 impl Client {
     #[wasm_bindgen(constructor)]
-    pub fn new(
-        // TODO: Change this to an options struct
-        instructions: String,
-        statement: String,
-        jwt: Option<String>,
-        ld: Option<String>,
-        verify_jwt: Option<String>,
-        verify_ld: Option<String>,
-    ) -> Result<Client, String> {
-        let jwt: Option<Url> = match jwt {
+    pub fn new(config: &str) -> Result<Client, String> {
```

Comment:
to take this a step further, you can use [serde_wasm_bindgen](https://docs.rs/serde-wasm-bindgen/0.5.0/serde_wasm_bindgen/) to remove the need for `JSON.stringify`ing the config on the JS side, just an observation tho